src/com/owncloud/android/lib/resources/shares/CreateRemoteShareOperation.java

```diff
@@ -141,45 +132,22 @@ protected RemoteOperationResult run(OwnCloudClient client) {
 			if(isSuccess(status)) {
 				String response = post.getResponseBodyAsString();
 
-				result = new RemoteOperationResult(ResultCode.OK);
-				
-				// Parse xml response --> obtain the response in ShareFiles ArrayList
-				// convert String into InputStream
-				InputStream is = new ByteArrayInputStream(response.getBytes());
-				ShareXMLParser xmlParser = new ShareXMLParser();
-				mShares = xmlParser.parseXMLResponse(is);
-				if (xmlParser.isSuccess()) {
-					if (mShares != null) {
-						Log_OC.d(TAG, "Created " + mShares.size() + " share(s)");
-						result = new RemoteOperationResult(ResultCode.OK);
-						ArrayList<Object> sharesObjects = new ArrayList<Object>();
-						for (OCShare share: mShares) {
-							sharesObjects.add(share);
-						}
-						result.setData(sharesObjects);
-
-						if (mGetShareDetails) {
-							// retrieve more info
-							OCShare emptyShare = (OCShare) sharesObjects.get(0);
-
-							GetRemoteShareOperation getInfo = new GetRemoteShareOperation(emptyShare.getIdRemoteShared());
-							result = getInfo.execute(client);
-						}
-
-					} else {
-						result = new RemoteOperationResult(ResultCode.UNKNOWN_ERROR);
-						Log_OC.e(TAG, "Successful response with no share in it");
+				ShareToRemoteOperationResultParser parser = new ShareToRemoteOperationResultParser(
+						new ShareXMLParser()
+				);
+				parser.setOneOrMoreSharesRequired(true);
+				parser.setOwnCloudVersion(client.getOwnCloudVersion());
+				parser.setServerBaseUri(client.getBaseUri());
+				result = parser.parse(response);
+
+				if (result.isSuccess()) {
+					Log_OC.d(TAG, "Created " + result.getData().size() + " share(s)");	// should be one
+					if (mGetShareDetails) {
```

Comment:
The two if statements could be merged into one with && except when the log statement is important to be done anyways not depending on the second condition.